DemoEvershop/PageObjects/DemoShopPage.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.Playwright;
+
+namespace DemoEverShop.PageObjects;
+
+internal class DemoShopPage(IPage page)
+{
+    private ILocator KidsCatalogLink => page.GetByRole(AriaRole.Link, new() { Name = "Shop kids" });
+    private ILocator AddToCartButton => page.GetByRole(AriaRole.Button, new() { Name = "ADD TO CART" });
+    private ILocator ViewCartButton => page.Locator(".add-cart-popup-button");
+    private ILocator CheckoutButton => page.GetByRole(AriaRole.Link, new() { Name = "CHECKOUT" });
+    private ILocator Loader => page.Locator(".loading-bar");
+    public ILocator CheckoutForm => page.Locator("#checkoutShippingAddressForm");
+
+    public async Task GoToDemoEverShopPage()
+    {
+        await page.GotoAsync(UiTestFixture.BaseUrl);
+    }
+    
+    public async Task OpenKidsShoesCatalog()
+    {
+        await KidsCatalogLink.WaitForAsync(new LocatorWaitForOptions { State = WaitForSelectorState.Visible });
+        await KidsCatalogLink.ClickAsync();
+    }
+    
+    public async Task OpenProductDetails(string productName)
+    {
+        await page.Locator($"//a/span[text()='{productName}']").WaitForAsync(new LocatorWaitForOptions { State = WaitForSelectorState.Visible });
+        await page.Locator($"//a/span[text()='{productName}']").ClickAsync();
+        await Task.Delay(1500);
```

Comment:
instead of hardcoding this delay, you can wait for colorOption to be visible before click it

Reply:
WaitForAsync and IsVisibleAsync don't work in this case. The element is visible and the test even does click on it but the element is not selected. Replaced with WaitForLoadStateAsync